### PR TITLE
Report item move

### PR DESCRIPTION
**Pull request by jennykbrennan**

### CODE REVIEW

**Comment by ScottChesworth** [comment]
Can anyone clue us in on how to find out why checks failed? Poking at the AppVeyor log here I only see scons exiting with error code 2, I'm not seeing why that happened.

**Comment by jcsteh** [comment]
I don't know why the Windows build isn't showing the error output, but the Mac output is more useful. (Wow, that's a first.)

```
duplicate symbol 'postMoveEnvelopePoint(int)' in:
    build/reaper_osara.os
    build/envelopeCommands.os
```

I haven't looked at your code, but it looks like postMoveEnvelopePoint somehow got defined in reaper_osara.cpp, when it's already defined in envelopeCommands.cpp.

**Comment by jennykbrennan** [comment]
Doh! Because that branch was a dud. Should have created a new one. Let's see if I can sort this out.

**Comment by jennykbrennan** [comment]
EIn other words. I experimented in reaper--osara cpp before I figured out how and where to do this. Totally slipped my mind. So, what's the easiest way to sort this out, I wonder.

**Comment by jennykbrennan** [comment]
I'll let the more learned have a peak at this for now and clean up as needed. . @jcsteh  I think I might want to try my hands on making this it's own function, as you described, but how opposed are you to keeping it working this way for now?

**Comment by jcsteh** [comment]
I'm really not trying to be difficult, but I don't want to merge this the way it is. There are other actions which use this function which aren't context sensitive, so this might misbehave for those.

**Comment by ScottChesworth** [comment]
We're gonna have a look at this tonight, hopefully will have a cleaner commit soon.

File: src/reaper_osara.cpp [hunk 1]
```diff
@@ -1718,6 +1718,37 @@ void postSelectMultipleItems(int command) {
...
+					translate_plural("{} item moved down", "{} items moved down", count),
+					count));
+				break;
+			}
+//			default: {
+//				outputMessage("Unknown case ");
+//			}
+		} 
+	}
+	else postMoveEnvelopePoint(command);
```
Comment (jcsteh) [inline src/reaper_osara.cpp:1749]: nit: else on same line as closing brace, avoid unbraced else block.
```suggestion
	} else {
		postMoveEnvelopePoint(command);
	}
```

File: src/reaper_osara.cpp [hunk 2]
```diff
@@ -1718,6 +1718,37 @@ void postSelectMultipleItems(int command) {
...
+void postMoveItemOrEnvelopePoint(int command) {
+	TrackEnvelope* envelope = GetSelectedEnvelope(nullptr);
+	if (!envelope) {
+		int count = CountSelectedMediaItems(0);
+		if(count==0) {
+			outputMessage(translate("no items selected"));
+			return;
+		}
+		// Translators: Reported when moving items{} will be replaced by the
+	// number of items; e.g. "2 items moved".
```
Comment (jcsteh) [inline src/reaper_osara.cpp:1730]: Unfortunately, you're going to need to put this translators comment (or a slight variation of it) above each message. Translators comments are only associated with the translatable string immediately following them. So, as this stands, the second message won't get a comment in the .po file.

```suggestion
		// Translators: Reported when moving items{} will be replaced by the
	// number of items; e.g. "2 items moved".
```

File: src/reaper_osara.cpp [hunk 3]
```diff
@@ -1718,6 +1718,37 @@ void postSelectMultipleItems(int command) {
...
+		switch(command) {
+			case 40117: {
+				outputMessage(format(
```
Comment (jcsteh) [inline src/reaper_osara.cpp:1733]:
```suggestion
				// Translators: Reported when moving items up. {} will be replaced by the
				// number of items; e.g. "2 items moved up".
				outputMessage(format(
```

File: src/reaper_osara.cpp [hunk 4]
```diff
@@ -1718,6 +1718,37 @@ void postSelectMultipleItems(int command) {
...
+	if (!envelope) {
+		int count = CountSelectedMediaItems(0);
+		if(count==0) {
+			outputMessage(translate("no items selected"));
+			return;
+		}
+		// Translators: Reported when moving items{} will be replaced by the
+	// number of items; e.g. "2 items moved".
+		switch(command) {
+			case 40117: {
```
Comment (jcsteh) [inline src/reaper_osara.cpp:1732]: Comment action name.

```suggestion
			case 40117: { // Item edit: Move items/envelope points up one track/a bit
```

File: src/reaper_osara.cpp [hunk 5]
```diff
@@ -1718,6 +1718,37 @@ void postSelectMultipleItems(int command) {
...
+				outputMessage(format(
+					translate_plural("{} item moved up", "{} items moved up", count),
+					count));
+				break;
+			}
+			case 40118: {
+				outputMessage(format(
```
Comment (jcsteh) [inline src/reaper_osara.cpp:1739]:
```suggestion
				// Translators: Reported when moving items down. {} will be replaced by the
				// number of items; e.g. "2 items moved".
				outputMessage(format(
```

File: src/reaper_osara.cpp [hunk 6]
```diff
@@ -1718,6 +1718,37 @@ void postSelectMultipleItems(int command) {
...
+		// Translators: Reported when moving items{} will be replaced by the
+	// number of items; e.g. "2 items moved".
+		switch(command) {
+			case 40117: {
+				outputMessage(format(
+					translate_plural("{} item moved up", "{} items moved up", count),
+					count));
+				break;
+			}
+			case 40118: {
```
Comment (jcsteh) [inline src/reaper_osara.cpp:1738]:
```suggestion
			case 40118: { // Item edit: Move items/envelope points down one track/a bit
```

File: src/reaper_osara.cpp [hunk 7]
```diff
@@ -1718,6 +1718,37 @@ void postSelectMultipleItems(int command) {
...
+				outputMessage(format(
+					translate_plural("{} item moved down", "{} items moved down", count),
+					count));
+				break;
+			}
+//			default: {
+//				outputMessage("Unknown case ");
+//			}
```
Comment (jcsteh) [inline src/reaper_osara.cpp:1746]: Avoid commented out code. However, for paranoia, we can add an assertion here which will show up in debug builds if this code sees an id it doesn't expect.

```suggestion
			default:
				assert(false);
```

**Review by jcsteh — APPROVED** [review]
Thanks. I've made a few small changes here.

File: src/reaper_osara.cpp (hunk 2 above)
Comment (jcsteh) [reply src/reaper_osara.cpp:1730]:
```suggestion
```

**Comment by jennykbrennan** [comment]
Ah, nice. Thanks. :)

**Pull request merged**